src/vsync.c

```diff
@@ -179,69 +142,63 @@ static int vsync_opengl_oml_wait(session_t *ps) {
 
 	glXGetSyncValuesOML(ps->dpy, ps->reg_win, &ust, &msc, &sbc);
 	glXWaitForMscOML(ps->dpy, ps->reg_win, 0, 2, (msc + 1) % 2, &ust, &msc, &sbc);
-
 	return 0;
 }
-
-#endif
-
-/// Function pointers to wait for VSync.
-int (*const VSYNC_FUNCS_WAIT[NUM_VSYNC])(session_t *ps) = {
-#ifdef CONFIG_VSYNC_DRM
-    [VSYNC_DRM] = vsync_drm_wait,
-#endif
-#ifdef CONFIG_OPENGL
-    [VSYNC_OPENGL] = vsync_opengl_wait,
-    [VSYNC_OPENGL_OML] = vsync_opengl_oml_wait,
-#endif
-};
-
-#ifdef CONFIG_OPENGL
-static void vsync_opengl_swc_deinit(session_t *ps) {
-	vsync_opengl_swc_swap_interval(ps, 0);
-}
 #endif
 
-/// Function pointers to deinitialize VSync.
-void (*const VSYNC_FUNCS_DEINIT[NUM_VSYNC])(session_t *ps) = {
-#ifdef CONFIG_OPENGL
-    [VSYNC_OPENGL_SWC] = vsync_opengl_swc_deinit,
-    [VSYNC_OPENGL_MSWC] = vsync_opengl_swc_deinit,
-#endif
-};
-
 /**
  * Initialize current VSync method.
  */
 bool vsync_init(session_t *ps) {
-	// Mesa turns on swap control by default, undo that
 #ifdef CONFIG_OPENGL
-	if (bkend_use_glx(ps))
+	if (bkend_use_glx(ps)) {
+		// Mesa turns on swap control by default, undo that
 		vsync_opengl_swc_swap_interval(ps, 0);
+	}
+#endif
+#ifdef CONFIG_VSYNC_DRM
+	log_warn("The DRM vsync method is deprecated, please don't enable it.");
```

Comment:
This warning is always shown, even when vsync is disabled. Is that on purpose?
I think with the new boolean option these warning don't need to be shown.

Reply:
This is intentional, users shouldn't enable the DRM vsync method (which is a build time option) unless there is really good reason.